telegram/ext/filters.py

```diff
@@ -192,12 +192,126 @@ def filter(self, message):
+        class file_type(BaseFilter):
+            """This Filter filters documents by their mime-type attribute
+
+            Note:
+                This Filter only filters by the mime_type of the document,
+                    it doesn't check the validity of document.
+                The user can manipulate the mime-type of a message and
+                    send media with wrong types that don't fit to this handler.
+
+            Examples:
+                Filters.documents.file_type("audio/mpeg") filters all audio in mp3 format.
+            """
+
+            def __init__(self, filetype):
+                """Initialize the category you want to filter
+
+                Args:
+                    filetype (str, optional): mime_type of the media you want to filter"""
+                self.filetype = filetype
+                self.name = "Filters.document.Filetype(\"{}\")".format(self.filetype)
+
+            def filter(self, message):
+                if message.document:
+                    return message.document.mime_type == self.filetype
+
+        apk = file_type("application/vnd.android.package-archive")
+        doc = file_type("application/msword")
+        docx = file_type("application/vnd.openxmlformats-officedocument.wordprocessingml.document")
+        exe = file_type("application/x-ms-dos-executable")
+        gif = file_type("video/mp4")
+        jpg = file_type("image/jpeg")
+        mp3 = file_type("audio/mpeg")
+        pdf = file_type("application/pdf")
+        py = file_type("text/x-python")
+        svg = file_type("image/svg+xml")
+        txt = file_type("text/plain")
+        targz = file_type("application/x-compressed-tar")
+        wav = file_type("audio/x-wav")
+        xml = file_type("application/xml")
+        zip = file_type("application/zip")
+
         def filter(self, message):
             return bool(message.document)
 
     document = _Document()
     """:obj:`Filter`: Messages that contain :class:`telegram.Document`."""
 
+    class file_size(BaseFilter):
+        """This Filter filters all messages with a `file_size` attribute."""
+
+        def __init__(self, min=None, max=None):
+            """Initialize the limits of the file_size in the `__init__` method.
+
+            Note:
+                If no limits are given, the filter returns `True` for every message
+                with file_size attritute
+
+            Args:
+                min (int, optional): Minimum `file_size` of the message media in Byte.
+                max (int, optional): Maximum `file_size` of the message media in Byte.
+            """
+            self.min = min
+            self.max = max
+            self.name = "Filters.file_size(min={0}, max={1})".format(self.min, self.max)
+
+        def filter(self, message):
+            if message.audio:
+                filesize = message.audio.file_size
+            elif message.document:
+                filesize = message.document.file_size
+            elif message.photo:
```

Comment:
`message.photo` is an array of different photo sizes. Not sure if the filter even makes here

Reply:
Oh, thanks. Didn't thought about that. What about just using the photo in the highest resolution?